src/routes/docs/tutorials/sveltekit-ssr-auth/step-4/+page.markdoc

```diff
@@ -0,0 +1,71 @@
+---
+layout: tutorial
+title: Add a server hook
+description: Add Authentication to a SvelteKit project using Appwrite.
+step: 4
+---
+
+SvelteKit hooks which are functions that run on the server before a page is displayed to the user. SvelteKit locals are a way to store data that is specific to the current request. We can use this to store the user's account data, so that it is available to all pages.
+
+Create a new file in the `src/hooks` directory called `server.js`:
+```js
+// src/hooks.server.js
+import { createAppwriteClient } from '$lib/server/appwrite';
+
+export async function handle({ event, resolve }) {
+	// Use our helper function to create the Appwrite client.
+	const { account } = createAppwriteClient(event);
+
+	// Let's store the current logged in user in locals,
+	// for easy access in our other routes.
+	try {
+		const user = await account.get();
+		event.locals.user = user;
+	} catch (error) {}
+
+	// Continue with the request.
+	return resolve(event);
+}
+```
+
+To ensure the `locals` object is typed correctly, we can add a type definition for it in the `env.d.ts` file:
+
+```ts
+import type { Models } from 'node-appwrite';
+import type { AppwriteService } from '$lib/server/appwrite';
+
+// See https://kit.svelte.dev/docs/types#app
+// for information about these interfaces
+declare global {
+	namespace App {
+		// interface Error {}
+		interface Locals {
+			user: Models.User<Preferences> | null;
+		}
+		// interface PageData {}
```

Comment:
Meant to be a comment?

Reply:
Yeah this is how sveltekit templates this files - best to keep this way I think

Reply:
Nope - this is how sveltekit templates the file. I think we should keep it this way